src/frontends/src/query_frontend.hpp

```diff
@@ -178,15 +178,15 @@ class RequestHandler {
               event_type);
           if (event_type == "set") {
             std::string model_name = key;
-            int new_version = clipper::redis::get_current_model_version(
+            std::string new_version = clipper::redis::get_current_model_version(
                 redis_connection_, key);
-            if (new_version >= 0) {
+            if (new_version.size() > 0) {
```

Comment:
I want `clipper::redis::get_current_model_version` to return a `boost::optional<std::string>` as noted in a comment below, so the correct check here is whether the `optional` contains a value.